xamarin/Xamarin.Forms
# [Android,iOS,UWP] Allow Entry CursorPosition/SelectionLength to be set in ctor

### Description of Change ###

#2539 added the ability to set `CursorPosition` and `SelectionLength` on `Entry`. It was set up to allow these properties to be set during construction, but that was thwarted by the initial `Text` triggering a `SelectionChanged`, which reverted the `CursorPosition` and `SelectionLength` to default values. This change established a one-time-use bool that will skip the initial `SelectionChanged` work.

Also resolved issue where clearing the set values of CursorPosition/SelectionLength did nothing.

### Issues Resolved ###

- fixes #3343 
- fixes #3633

### API Changes ###

None

### Platforms Affected ###

- iOS
- Android
- UWP

### Behavioral/Visual Changes ###

None

### Testing procedure ###

1. Open test case 3343.
2. First Entry should have selection. CursorPositon: 4 SelectionLength: 7.
3. Second Entry should have no selection. CursorPostion: 0 SelectionLength: 0.
4. Tap in second Entry. Make selection. CursorPosition and SelectionLength should update accordingly. Note: iOS will snap to end on first click, tap and hold to move the cursor.
5. Tap quickly between the two entries. Should not crash.
6. Tap Click Me button. 
7. Second Entry should have selection. CursorPosition: 4. SelectionLength: 8.
8. Tap Click Me After.
9. Second Entry should have selection. CursorPosition: 2 SelectionLength 8.
10. Tap Click Me Last. Watch the second entry.
11. Second Entry selection should disappear. CursorPosition: 2 SelectionLength 0. 
12. Second Entry CursorPostion should snap to end. CursorPostion: 12 SelectionLength 0.
13. Clear the text in the second Entry. CursorPosition: 0 SelectionLength: 0
14. Tap Click Me. Should not crash. CursorPosition: 0 SelectionLength: 0
15. Type three characters in the second Entry. CursorPosition: 3 SelectionLength: 0
16. Tap Click Me. Should not crash. CursorPosition: 3 SelectionLength: 0
17. Tap Click Me After. Should not crash. CursorPosition: 2 SelectionLength: 0
18. Repeat 15-17 with varying text lengths.

### PR Checklist ###
